framework/common/src/main/java/edu/kit/kastel/mcse/ardoco/core/api/textextraction/NounMappingChangeListener.java

```diff
@@ -1,6 +1,8 @@
-/* Licensed under MIT 2022-2023. */
+/* Licensed under MIT 2022-2024. */
 package edu.kit.kastel.mcse.ardoco.core.api.textextraction;
 
-public interface NounMappingChangeListener {
+import java.io.Serializable;
+
+public interface NounMappingChangeListener extends Serializable {
```

Comment:
Yes, since we do not necessarily know when a state is serialized, we want to preserve relationships as best as we can. Imagine a scenario like the Inconsistency Detection, where we want to run the same stage X times using the same "base run"/"pre run". To achieve proper separation between runs using a deep copy is required, which necessitates serializability of all things belonging to a state, including relationships between state elements.